transifex/pull.py

```diff
@@ -288,12 +297,28 @@ def pull(repo):
         repo.cleanup(pr)
 
 
+def parse_arguments():
+    parser = ArgumentParser()
+    parser.add_argument(
+        'clone_url',
+        nargs='?',
+        help='URL to use to clone the repository. If blank, URLs will be pulled from settings.yaml'
+    )
+    parser.add_argument(
+        '--skip-compile-messages',
+        action='store_true',
+        help='Skip the message compilation step.'
+    )
+    return parser.parse_args()
+
+
 if __name__ == '__main__':
-    try:
-        clone_url = sys.argv[1]
-        repo = Repo(clone_url)
-        pull(repo)
-    except IndexError:
+    args = parse_arguments()
+
+    if args.clone_url:
+        repo = Repo(args.clone_url)
+        pull(repo, skip_compile_messages=args.skip_compile_messages)
```

Comment:
I think I'd prefer to leave it as is, as I suspect I will be refactoring a lot of the Repo code when I write the push script, as much of it will be shared between the two files.